falcon/util/misc.py

```diff
@@ -77,6 +78,22 @@ def decorator(func):
     _lru_cache_safe = functools.lru_cache
 
 
+def is_cython_func(func):
+    """Determines whether a function is implemented with Cython.
+
+    Args:
+        func: The function object to check.
+    Returns:
+        bool: ``True`` if the function is a Cython function or method;
+        ``False`` otherwise.
+
+    """
+    if inspect.ismethod(func):
+        func = func.__func__
+
+    return 'cython' in type(func).__name__
```

Comment:
I'm not getting this heuristic to work for functions (as opposed to cythonized class methods), they just show up as builtins:
```python
>>> import falcon
>>> falcon.uri.parse_query_string
<built-in function parse_query_string>
>>> falcon.API
<class 'falcon.app.API'>
>>> falcon.API.__init__
<cyfunction API.__init__ at 0x7f8818024608>
```

Reply:
Hmm, I could have sworn that was working at some point but I must have been mistaken. I'll take a look.

Reply:
I was able to repro this after adding some more tests to `test_cythonized_asgi`. Working on a fix.